src/Settings/Default.hs

```diff
@@ -215,6 +215,44 @@ defaultBuilderArgs = mconcat
     , makeBuilderArgs
     , tarBuilderArgs ]
 
+-- | Disable some warnings in packages we use
+-- | https://github.com/ghc/ghc/blob/master/mk/warnings.mk#L46
+disableWarningArgsStage0 :: Args
+disableWarningArgsStage0 = stage Stage0 ? builder Ghc ? mconcat
+    [ package transformers ? mconcat [ arg "-fno-warn-unused-matches", arg "-fno-warn-unused-imports" ]
+    , package terminfo ? mconcat [ arg "-fno-warn-unused-imports" ] ]
+
+-- XXX: How to express Stage1 and upper?
+-- XXX: What about the commented-out packages?
```

Comment:
There are a few packages which are not built by the Make build system by default and which I therefore never tried to support. I think I would prefer to leave them out for now for two reasons:
* Our priority is to build GHC and the default set of packages as fast as we can.
* Once Hadrian is in the tree, adding a new package (e.g. `vector`) should be a relatively straightforward and well-defined task, which could be handled in a separate GHC ticket if/when needed.